src/services/vector_store_service.py

```diff
@@ -0,0 +1,122 @@
+"""
+VectorStoreService - Responsible for vector database operations.
+Handles creation of vector stores and similarity searches.
+"""
+
+import re
+import logging
+import faiss
+from typing import List, Dict, Optional, Tuple
+from langchain_community.vectorstores import FAISS
+from langchain_community.docstore.in_memory import InMemoryDocstore
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+
+from Utils.embedding_utils import check_text_size_before_embedding
+from common.constants import REGEX_PATTERNS
+
+logger = logging.getLogger(__name__)
```

Comment:
Any particular reason why it is defined outside the class scope?

Reply:
all of our modules follow same pattern we are definig loggers always in the module level 

what wrong thing do you see exactly of not to keep following this pattern in this particular module?